astropy/modeling/fitting.py

```diff
@@ -549,24 +549,32 @@ def __call__(self, model, x, y, z=None, weights=None, **kwargs):
         fitted_model = self.fitter(model, x, y, z, weights=weights, **kwargs)
         if z is None:
             filtered_data = y
+            filtered_weights = weights
             for n in range(self.niter):
                 filtered_data = self.outlier_func(filtered_data - fitted_model(x),
                                                   **self.outlier_kwargs)
                 filtered_data += fitted_model(x)
+                if weights is not None:
+                    filtered_weights = weights[~filtered_data.mask]
                 fitted_model = self.fitter(fitted_model,
                                x[~filtered_data.mask],
                                filtered_data.data[~filtered_data.mask],
+                               weights= filtered_weights,
                                **kwargs)
         else:
             filtered_data = z
+            filtered_weights = weights
             for n in range(self.niter):
                 filtered_data = self.outlier_func(filtered_data - fitted_model(x, y),
                                                   **self.outlier_kwargs)
                 filtered_data += fitted_model(x, y)
+                if weights is not None:
+                    filtered_weights = weights[~filtered_data.mask]
                 fitted_model = self.fitter(fitted_model,
                                x[~filtered_data.mask],
                                y[~filtered_data.mask],
                                filtered_data.data[~filtered_data.mask],
+                               weights= filtered_weights,
```

Comment:
Could you remove the white space right after ``=``?

Reply:
sure